sui_core/src/safe_client.rs

```diff
@@ -263,8 +265,52 @@ where
     /// Handle Batch information requests for this authority.
     async fn handle_batch_streaming(
         &self,
-        _request: BatchInfoRequest,
+        request: BatchInfoRequest,
     ) -> Result<Receiver<Result<BatchInfoResponseItem, SuiError>>, io::Error> {
-        todo!()
+        let (mut tx_output, tr_output) = channel(BUFFER_SIZE);
+
+        let mut batch_info_items = self
+            .authority_client
+            .handle_batch_streaming(request)
+            .await?;
+
+        for next_data in batch_info_items.next().await {
+            match next_data {
+                Ok(batch_info_response_item) => {
+                    // do security checks
+                    match batch_info_response_item.clone() {
+                        BatchInfoResponseItem(UpdateItem::Batch(signed_batch)) => {
+                            // check signature of batch
+                            let result = signed_batch.signature.check(&signed_batch.signature, signed_batch.authority);
+                            match result {
+                                Ok(_) => {
+                                    let _ = tx_output.send(Ok(batch_info_response_item)).await;
+                                },
+                                Err(e) => {
+                                    let _ = tx_output.send(Err(e)).await;
+                                },
+                            }
+                        }
+                        BatchInfoResponseItem(UpdateItem::Transaction((_seq, digest))) => {
+                            // make transaction info request which checks transaction certificate
+                            let transaction_info_request = TransactionInfoRequest{ transaction_digest: digest };
+                            let transaction_info_response = self.handle_transaction_info_request(transaction_info_request).await;
```

Comment:
This actually does not pipeline the requests, so it will be slow. We should fix the underlying implementation (best, todo) or make a pipelined variant for this use case.

Reply:
Made an issue for this, and will work on this shortly https://github.com/MystenLabs/sui/issues/1127

Reply:
Right now, lets assume you need to download them and finalize this PR, we can later specify a flag on whether the requester wants to request them by default or not.